### PR TITLE
[Audio Processor] Only pass sr to feat extractor

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
# What does this PR do?

The audio processor is composed of two components:
1. Feature extractor (input audio -> normalised audio)
2. Tokenizer (target text-> label ids)

Of these two components, the `audio` inputs and `sampling_rate` are arguments that are applicable to the feature extractor only. The `text` is applicable to the tokenizer only.

Currently, we only isolate the `audio` for the feature extractor and `text` for the tokenizer. However, the `sampling_rate` is passed to **both** the feature extractor and tokenizer. Thus, we get a warning for an unrecognized keyword argument in the tokenizer:

```python
from transformers import Wav2Vec2Processor
import numpy as np

audio = np.ones((2, 1000))
text = ['the cat', 'sat on']

processor = Wav2Vec2Processor.from_pretrained('facebook/wav2vec2-base')

out = processor(audio, sampling_rate=16000, text=text)
```

```
Keyword arguments {'sampling_rate': 16000} not recognized.
```

This PR splits the `sampling_rate` from the kwargs and passes it only to the feature extractor.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by sanchit-gandhi** [comment]
Currently, I've only applied the change to the Wav2Vec2 Processor - once we're happy with the change I'll copy it to all audio processor classes. I've only done it to this one first to make the review easier!

File: src/transformers/models/wav2vec2/processing_wav2vec2.py [hunk 1]
```diff
@@ -78,8 +78,10 @@ def __call__(self, *args, **kwargs):
         if "raw_speech" in kwargs:
             warnings.warn("Using `raw_speech` as a keyword argument is deprecated. Use `audio` instead.")
             audio = kwargs.pop("raw_speech")
+            sampling_rate = kwargs.pop("sampling_rate", None)
         else:
             audio = kwargs.pop("audio", None)
+            sampling_rate = kwargs.pop("sampling_rate", None)
```
Comment (sgugger) [inline src/transformers/models/wav2vec2/processing_wav2vec2.py:84]: Since it's there in both path, you should just put it once outside of the if ;-)

**Review by sgugger — APPROVED** [review]
Looks like a welcome change, thanks!

File: src/transformers/models/wav2vec2/processing_wav2vec2.py (hunk 1 above)
Comment (sanchit-gandhi) [reply src/transformers/models/wav2vec2/processing_wav2vec2.py]: Whoops 😅 you're entirely correct! Resolved in [5568d32](https://github.com/huggingface/transformers/pull/20022/commits/5568d32eb0823a029117c0067e33cd7093f2fe9d)

**Pull request merged**